pkg/controllers/streaming/processor_controller.go

```diff
@@ -18,36 +18,506 @@ package streaming
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
+	"github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	"github.com/projectriff/system/pkg/tracker"
+
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
 	streamingv1alpha1 "github.com/projectriff/system/pkg/apis/streaming/v1alpha1"
 )
 
 // ProcessorReconciler reconciles a Processor object
 type ProcessorReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log       logr.Logger
+	Scheme    *runtime.Scheme
+	Tracker   tracker.Tracker
+	Namespace string
 }
 
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=build.projectriff.io,resources=functions,verbs=get;watch
+// +kubebuilder:rbac:groups=streaming.projectriff.io,resources=streams,verbs=get;watch
+// +kubebuilder:rbac:groups=keda.k8s.io,resources=scaledobject,verbs=get;watch
 
 func (r *ProcessorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("processor", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("processor", req.NamespacedName)
+
+	var original streamingv1alpha1.Processor
+	if err := r.Client.Get(ctx, req.NamespacedName, &original); err != nil {
+		return ctrl.Result{}, ignoreNotFound(err)
+	}
+
+	// Don't modify the informers copy
+	processor := original.DeepCopy()
+
+	// Reconcile this copy of the processor and then write back any status
+	// updates regardless of whether the reconciliation errored out.
+	result, err := r.reconcile(ctx, log, processor)
+
+	// check if status has changed before updating, unless requeued
+	if !result.Requeue && !equality.Semantic.DeepEqual(original.Status, processor.Status) {
+		if updateErr := r.Status().Update(ctx, processor); updateErr != nil {
+			log.Error(updateErr, "unable to update Processor status")
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+	}
+	return result, err
+}
+
+func (r *ProcessorReconciler) reconcile(ctx context.Context, logger logr.Logger, processor *streamingv1alpha1.Processor) (ctrl.Result, error) {
+	if processor.GetDeletionTimestamp() != nil {
+		return ctrl.Result{}, nil
+	}
+
+	// We may be reading a version of the object that was stored at an older version
```

Comment:
for consistency with other controllers, move this line through `InitializeConditions()` out to the main Reconcile method just below `processor := original.DeepCopy()`.